libflux/core/src/parser/mod.rs

```diff
@@ -1597,12 +1600,14 @@ impl Parser {
                     rbrack: self.make_comments(&end),
                 }))
             }
-            _ => {
+            t => {
                 let comma = self.expect(TokenType::Comma);
                 let mut items = vec![ArrayItem {
                     expression: init,
                     comma: self.make_comments(&comma),
                 }];
+                // keep track of the last token's byte offsets
+                let mut last = (t.start_offset, t.end_offset);
```

Comment:
It should be sufficient to use only `start_offset`. There shouldn't be a case where only `end_offset` has changed. You can just go `self.peek().start_offset`